File: main/src/main/java/net/fwitz/math/main/binary/dual/functions/examples/simple/DualInverseInverse.java
package net.fwitz.math.main.binary.dual.functions.examples.simple;

import net.fwitz.math.plot.binary.dual.DualNumberFunctionPlot;

public class DualInverseInverse {
    public static void main(String[] args) {
        new DualNumberFunctionPlot("1 / (1/z)")
                .computeFn(z -> z.inverse().inverse())
                .render();
    }
}

File: main/src/main/kotlin/net/fwitz/math/main/binary/dual/functions/examples/simple/DualInverseInverse.kt
package net.fwitz.math.main.binary.dual.functions.examples.simple

import net.fwitz.math.plot.binary.dual.DualNumberFunctionPlot

object DualInverseInverse {
    @JvmStatic
    fun main(args: Array<String>) = DualNumberFunctionPlot("1 / (1/z)")
        .computeFn { z -> z.inverse.inverse }
        .render()
}